### PR TITLE
Adding Release GitHub Action

### PR DESCRIPTION
**Pull request by nniehoff**
This PR introduces a GitHub Action to publish Nautobot python packages to GitHub and PyPi as well as build and push the docker containers.  I added a slack notification as well which we can customize or remove as needed.  Also, I need someone on the core team to validate the secrets we are using here and let me know if I need to change them.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,138 @@
+---
+name: "Release"
+on:
+  - "release"
```
Comment (nniehoff) [inline .github/workflows/release.yml:4]: Do we want to limit the type of release to execute this action on?  https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#release

**Comment by nniehoff** [comment]
This might be a bit premature with https://github.com/nautobot/nautobot/pull/1184 on the way but I figure this is the best we have now and matches the current process.  Maybe this will save some time.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+          file_glob: true
+      - name: "Push to PyPI"
+        uses: "pypa/gh-action-pypi-publish@release/v1"
+        with:
+          user: "__token__"
+          password: "${{ '{{' }} secrets.PYPI_API_TOKEN {{ '}}' }}"
+
+  container-publish:
+    runs-on: "ubuntu-20.04"
+    if: "github.event_name == 'push' && (github.ref == 'refs/heads/develop' || github.ref == 'refs/heads/next')"
```
Comment (jathanism) [inline .github/workflows/release.yml:45]: We want to include `main`!

File: .github/workflows/release.yml (hunk 1 above)
Comment (jathanism) [reply .github/workflows/release.yml:4]: Let's just scope it to `published`:

```
on:
  release:
    types: [published]
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        with:
+          python-version: "3.9"
+      - name: "Install Python Packages"
+        run: "pip install poetry"
+      - name: "Run Poetry Build"
+        run: "poetry build"
+      - name: "Upload binaries to release"
+        uses: "svenstaro/upload-release-action@v2"
+        with:
+          repo_token: "${{ '{{' }} secrets.NTC_GITHUB_TOKEN {{ '}}' }}"
```
Comment (jathanism) [inline .github/workflows/release.yml:32]:
```suggestion
          repo_token: "${{ '{{' }} secrets.GITHUB_TOKEN {{ '}}' }}"
```

**Review by jathanism — CHANGES_REQUESTED** [review]
Secrets needed to be added to the repo for this to work:

- secrets.PYPI_API_TOKEN
- secrets.SLACK_WEBHOOK_URL
- secrets.DOCKERHUB_USERNAME
- secrets.DOCKERHUB_TOKEN

**Review by jathanism — APPROVED** [review]

**Pull request merged**